### DISCUSSION TITLE
How to update an item in a list attribute?

### DISCUSSION THREAD

**Discussion by barticus** [sensedeep/dynamodb-onetable#456] [Q&A] [answered]
Hi,
I have a model with a list attribute. I'm trying to update the element at a specific index.
I understand the dynamo way to do this is to do something like this:

```
aws dynamodb update-item \
    --table-name ProductCatalog \
    --key '{"Id":{"N":"789"}}' \
    --update-expression "SET RelatedItems[1] = :ri" \
    --expression-attribute-values file://values.json \
    --return-values ALL_NEW
```

([from docs](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.UpdateExpressions.html#Expressions.UpdateExpressions.SET))

But I'm struggling to work out the set syntax for that. I tried the below, but get the error `OneTable execute failed "update" for "ScheduledEnvironment. The provided key element does not match the schema` 

(code adapted to match the above)

```
ProductCatalog.update(
    {
      Id: 789,
    },
    {
      set: {
        [`RelatedItems[${index}]`]: "@{item}",
      },
      substitutions: {
        item,
      },
    }
  );
```

**Comment by barticus** [comment, marked as answer]
Disregard, I realised the error was with the SK not resolving correctly.
The format I originally posted works, for anyone else looking into this.

**Reply by dev-embedthis** [reply]
Thanks for following up and clarifying.
